gms/factories/src/main/java/com/linkedin/common/factory/KafkaEventProducerFactory.java

```diff
@@ -23,11 +25,18 @@
   private String kafkaSchemaRegistryUrl;
 
   @Bean(name = "kafkaEventProducer")
-  protected Producer<String, IndexedRecord> createInstance() {
-    Properties props = new Properties();
-    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers);
-    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
-    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class.getName());
+  protected Producer<String, IndexedRecord> createInstance(KafkaProperties properties) {
+    KafkaProperties.Producer producerProps = properties.getProducer();
+
+    producerProps.setKeySerializer(StringSerializer.class);
+    producerProps.setValueSerializer(KafkaAvroSerializer.class);
+
+    // KAFKA_BOOTSTRAP_SERVER has precedence over SPRING_KAFKA_BOOTSTRAP_SERVERS
+    if (kafkaBootstrapServers != null && kafkaBootstrapServers.length() > 0) {
```

Comment:
`kafkaBootstrapServers` will never be null I guess because it has a default set as in line 21? Maybe, we can get rid of the default value in this case if `KafkaProperties` defaults to `localhost:9092`?

Reply:
Nice find; I've removed the default.